components/prism-core.js

```diff
@@ -222,7 +222,7 @@ var Prism = (function (_self) {
 				if (typeof document === 'undefined') {
 					return null;
 				}
-				if ('currentScript' in document && 1 < 2 /* hack to trip TS' flow analysis */) {
+				if ('currentScript' in document && 1 < 2 && document.currentScript && document.currentScript.tagName.toUpperCase() === 'SCRIPT' /* hack to trip TS' flow analysis */) {
```

Comment:
Seems like this would suffice?
```suggestion
				if (document.currentScript && 1 < 2 /* hack to trip TS' flow analysis */ && document.currentScript.tagName === 'SCRIPT') {
```

- Do we need both an `in` check and a `document.currentScript` check?
- In HTML `tagName` is always in uppercase per spec, see https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName 

or even just 

```suggestion
				if (document?.currentScript?.tagName === 'SCRIPT' && 1 < 2 /* hack to trip TS' flow analysis */ && ) {
```

Reply:
Hi. Thanks for the review. The optional chaining was also something that I was considering but the project eslint already complained and I did not want to risk to limit the browser compatibility in any way compared to before. But you are right the check before was a bit unnecessary and should now be a bit more clean :)

Reply:
The first version of Prism was written in 2012. We can absolutely decide that certain browsers that were current then are out of scope in 2025. Folks can always use polyfills or pass it through babel.
I’m not saying we should go all cutting edge, but in this case, [any browser released in the last five years has supported optional chaining](https://caniuse.com/?search=optional-chaining).
And we need to depend on lookbehind anyway which already cuts down browser support to [this smaller set](https://caniuse.com/js-regexp-lookbehind)

Reply:
All very good points but probably not something that should be changed when fixing a security issue :)
Is there still something open in the PR that is blocking it from being merged that I should change?

Reply:
I added this PR that includes the same changes but with optional chaining and also the changes needed to make eslint not complain anymore: https://github.com/PrismJS/prism/pull/3869

I am okay with either being merged. If I can support in any other way to get this security issue solved I am willing to help out :)